catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/EntityResourceTest.java

```diff
@@ -779,6 +779,29 @@ public void delete_nonExistentEntity_404() {
     assertResponse(exception, NOT_FOUND, entityNotFound(entityName, NON_EXISTENT_ENTITY));
   }
 
+  @Test
+  public void delete_put_Entity_200(TestInfo test) throws URISyntaxException, IOException {
+
+    // Create Entity
+    Object request = createRequest(getEntityName(test), "description", "displayName", null);
+    T entity = createAndCheckEntity(request, adminAuthHeaders());
+    EntityInterface<T> entityInterface = getEntityInterface(entity);
+
+    // Delete Entity
+    deleteEntity(entityInterface.getId(), adminAuthHeaders());
+
+    // PUT Entity back
+    ChangeDescription change = getChangeDescription(entityInterface.getVersion());
+    change.getFieldsUpdated().add(new FieldChange().withName("deleted").withNewValue(false).withOldValue(true));
+
+    // Validate version and change description
+    T updated = updateEntity(request, OK, adminAuthHeaders());
+    EntityInterface<T> updatedInterface = getEntityInterface(updated);
+    assertEquals(updatedInterface.getVersion(), entityInterface.getVersion() + 1.0); // major version update
+    assertEquals(updatedInterface.getChangeDescription().getFieldsUpdated(), change.getFieldsUpdated());
+
+  }
```

Comment:
Can we also add another few lines which would trigger a minor version update to the entity and assert that is the case?

Reply:
makes sense, will do 🚀  thanks!